content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md

```diff
@@ -48,33 +48,29 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
    To prevent SQL injection attacks, avoid concatenating unsafe user input with queries.
    {{% /warn %}}
 
-   ```js
-   const fluxQuery =
-     'from(bucket: "YOUR_BUCKET")
-      |> range(start: 0)
-      |> filter(fn: (r) => r._measurement == "temperature")'
-   ```
+    ```js
+    const fluxQuery =
+      'from(bucket: "YOUR_BUCKET")
+        |> range(start: 0)
+        |> filter(fn: (r) => r._measurement == "temperature")'
+    ```
    Replace *`YOUR_BUCKET`* with the name of your InfluxDB bucket.
 
-4. Use the `queryRows()` method of the query client to query InfluxDB.
-   `queryRows()` takes a Flux query and an [RxJS **Observer**](http://reactivex.io/rxjs/manual/overview.html#observer) object.
-   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an  [RxJS **Observable**](http://reactivex.io/rxjs/manual/overview.html#observable).
-   `queryRows()` subscribes your observer to the observable.
-   Finally, the observer logs the rows from the response to the terminal.
-
-   ```js
-   const observer = {
-     next(row, tableMeta) {
-       const o = tableMeta.toObject(row)
-       console.log(
-         `${o._time} ${o._measurement} in '${o.location}' (${o.sensor_id}): ${o._field}=${o._value}`
-       )
-     }
-   }
-
-   queryApi.queryRows(fluxQuery, observer)
-
-   ```
+4. Use the `iterateRows()` method of the query client to query InfluxDB.
+   `iterateRows()` takes a Flux query and returns table as an asynchronous collection.
+   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an as an AsyncIterable.
+
+  ```js
+  const myQuery = () => {
```

Comment:
This should an async lambda with proper error handling, don't you think?

Reply:
Users should learn about error handling in general js tutorials, so for this example is error handling just added unnecessary complexity.
Missing async keyword is mistake indeed.

Reply:
Uncaught promise rejection terminates the node.js process, that's why I think it is rather a good practice to let inexperienced users to learn from good examples. I wanted to let know about the facts and let others decide, I am not an approver herein ... also, the previous version was even worse, it was not correct about using observables at all.